sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcArrayColumnVector.java

```diff
@@ -31,26 +32,27 @@
  */
 public class OrcArrayColumnVector extends OrcColumnVector {
   private final OrcColumnVector data;
-  private final long[] offsets;
-  private final long[] lengths;
+  private ListColumnVector listData;
 
   OrcArrayColumnVector(
```

Comment:
I think we don't need to store another copy of `vector` here. We can change [`OrcColumnVector.baseData`](https://github.com/apache/spark/blob/master/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnVector.java#L32) from `private` to `protected`, and just use `baseData` to get offset and length. So how about defining the constructor like below:

```java
OrcArrayColumnVector(
    DataType type,
    ListColumnVector vector,
    OrcColumnVector data) {

  super(type, vector);

  this.data = data;
}
```